bigtable/admin.go

```diff
@@ -275,6 +278,47 @@ func (ac *AdminClient) CreateColumnFamily(ctx context.Context, table, family str
 	return err
 }
 
+// TableConf contains all of the information necessary to create a table with column families.
+type UpdateTableConf struct {
+	TableID string
+	// Set to true to make the table protected against data loss
+	// i.e. deleting the table, the column families in the table and the instance containing the table would be prohibited
+	DeletionProtection *bool
+}
+
+// UpdateTable updates a table in the instance from the given configuration.
+// Only deletion protection can be updated at this period.
+func (ac *AdminClient) UpdateTable(ctx context.Context, conf *UpdateTableConf) (updated bool, err error) {
+	if conf.TableID == "" {
+		return false, errors.New("TableID is required")
+	}
+
+	if conf.DeletionProtection != nil {
+		ctx = mergeOutgoingMetadata(ctx, ac.md)
+
+		updateMask := &field_mask.FieldMask{}
+		updateMask.Paths = append(updateMask.Paths, "deletion_protection")
+
+		req := &btapb.UpdateTableRequest{
+			Table: &btapb.Table{
+				Name:               conf.TableID,
+				DeletionProtection: *conf.DeletionProtection,
+			},
+			UpdateMask: updateMask,
+		}
+		lro, err := ac.tClient.UpdateTable(ctx, req)
+		if err != nil {
+			return false, err
+		}
+		err = longrunning.InternalNewOperation(ac.lroClient, lro).Wait(ctx, nil)
```

Comment:
Do we usually wait here or return the operation back?

Reply:
nvm, we are doing this in file. It is as not flexible as return the long running operation itself, but I okay with this.